src/components/AmountTextInput.js

```diff
@@ -30,36 +47,52 @@ class AmountTextInput extends React.Component {
   onChangeText = (text) => {
     if (text === '') {
       // Need to handle empty string separately
-      this.props.onAmountUpdate(text);
+      this.setState({ text: '' });
+      this.props.onAmountUpdate(text, null);
       return;
     }
 
     let parsedText = text;
+    let bigIntValue;
     if (this.props.allowOnlyInteger) {
       // We allow only integers for NFT
       parsedText = parsedText.replace(/[^0-9]/g, '');
     }
 
     parsedText = getAmountParsed(parsedText);
 
-    if (!this.props.allowOnlyInteger) {
-      const amount = getIntegerAmount(parsedText);
-      if (Number.isNaN(amount) || amount < 0) {
-        return;
+    // There is no NaN in BigInt, it either returns a valid bigint or throws
+    // an error.
+    let isValid = true;
+    try {
+      bigIntValue = getIntegerAmount(parsedText);
+
+      if (!this.props.allowOnlyInteger) {
+        if (bigIntValue < 0n) {
+          isValid = false;
+        }
       }
```

Comment:
Does it make sense to validate non negative values even if `this.props.allowOnlyInteger === true`?

Reply:
Yes it does, refactored.